node/make.js

```diff
@@ -59,7 +52,150 @@ target.loc = function() {
         }
     }
 
-    // Create the en-US resjson file.
+    // create the en-US resjson file.
     var enContents = JSON.stringify(strings, null, 2);
     fs.writeFileSync(path.join(strPath, 'resources.resjson'), enContents)
-}
\ No newline at end of file
+}
+
+var rp = function (relPath) {
+    return path.join(__dirname, relPath);
+}
+
+var run = function(cl) {
+    console.log('> ' + cl);
+    var rc = exec(cl).code;
+    if (rc !== 0) {
+        echo('Exec failed with rc ' + rc);
+        exit(rc);
+    }
+}
+
+var getExternals = function () {
+    // determine the platform
+    var platform = os.platform();
+    if (platform != 'darwin' && platform != 'linux' && platform != 'win32') {
+        throw new Error('Unexpected platform: ' + platform);
+    }
+
+    // download the same version of node used by the agent
+    // and add node to the PATH
+    var nodeUrl = 'https://nodejs.org/dist';
+    var nodeVersion = 'v5.10.1';
+    switch (platform) {
+        case 'darwin':
+            var nodeArchivePath = downloadArchive(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-darwin-x64.tar.gz');
+            addPath(path.join(nodeArchivePath, 'node-' + nodeVersion + '-darwin-x64', 'bin'));
+            break;
+        case 'linux':
+            var nodeArchivePath = downloadArchive(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-linux-x64.tar.gz');
+            addPath(path.join(nodeArchivePath, 'node-' + nodeVersion + '-linux-x64', 'bin'));
+            break;
+        case 'win32':
```

Comment:
Windows needs an extra step to copy node.exe and node.lib into a directory that can be added to the PATH. On OSX/Linux, the extra step isn't needed because we download and extract a targz. On Windows, the individual files are downloaded like this:
```
C:\vsts-task-lib\node\_download\file\https___nodejs.org_dist_v5.10.1_win-x64_node.exe
C:\vsts-task-lib\node\_download\file\https___nodejs.org_dist_v5.10.1_win-x64_node.lib
```